tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs

```diff
@@ -725,6 +734,7 @@ async Task<string> GenerateAsync (string projectName, string registerPath, (stri
 			using (var reader = new StreamReader(templatePath)) {
 				var result = await reader.ReadToEndAsync ();
 				result = result.Replace (DownloadPathKey, downloadPath);
+				result = result.Replace (ProjectGuidKey, GetProjectGuid (projectName));
```

Comment:
There's already `Harness.GetStableGuid`: https://github.com/xamarin/xamarin-macios/blob/9168a63698aaeb3d32488a32b4f4168c11cebd00/tests/xharness/Harness.cs#L747-L758

Reply:
@rolfbjarne is that method going to generate the Guid based on the project name. I fear that this method will depend on the order of the call, is that correct?

Reply:
Yes, `NewStableGuid` will return different values in subsequent calls. However, it will always return the same value for call #X (since the initial random seed is fixed). If csproj generation is always done in the same order, the returned GUIDs won't change.

If you really want to use the project name as input to get a stable GUID, I think you should augment the existing `NewStableGuid` method (maybe add a `string seed = null` argument, which uses your implementation if provided).

Reply:
@rolfbjarne I prefer to be able to pass the seed, since I'm sure people will edit the project definitions more ofter than other parts.. I might need to do some refactoring since the importer code does not know about Harness, but I prefer to reuse code.